pkg/operators/cli/clioperator.go

```diff
@@ -276,9 +310,15 @@ func (o *cliOperatorInstance) PreStart(gadgetCtx operators.GadgetContext) error
 			fields, hasFields = fieldLookup[""] // fall back to default
 		}
 
-		mode := o.mode
-		if annotatedMode := ds.Annotations()["cli.output"]; annotatedMode != "" {
-			mode = annotatedMode
+		mode, ok := modes[ds.Name()]
+		if !ok {
+			mode, ok = modes[""]
```

Comment:
It's because output can be now specified either as `columns` (Stored in the map with key `""`) or `datasource:json,datasource2:columns`. This is the same format we use for params like `map-fetch-interval` and `map-fetch-count` and helper `GetStringValuesPerDataSource` to parse it.

This support will have no impact on users because we don't have gadgets with more than one datasource. However, this code prepares things for that.

BTW, I improved the flag description to make it clearer: `Specifies output mode for all ("mode1") or per data source ("datasource:mode1,datasource2:mode2")`

Reply:
It's a map, so `""` is the key and `"columns"` is the value. When multiple datasource are specified, `"<datasource>"` is the key and `"columns"`, or `"json"` or whatever is the value. I'm sorry but don't get your point.